packages/fxa-auth-server/lib/payments/capability.ts

```diff
@@ -112,32 +114,27 @@ export class CapabilityService {
       return;
     }
 
-    // Stripe subscriptions from events do not have product expanded, we filter
-    // by product being the non-expanded string for type checks.
-    const affectedProductIds = sub.items.data
-      .map((item) => item.price.product)
-      .filter((product) => typeof product === 'string') as string[];
-    if (affectedProductIds.length === 0) {
+    // Stripe subscriptions from events do not have price expanded, we filter
+    // by price being the non-expanded string for type checks.
+    const affectedPriceIds = sub.items.data.map((item) => item.price.id);
+    if (affectedPriceIds.length === 0) {
       return;
     }
-    const currentProductIds = await this.subscribedProductIds(uid);
-    let priorProductIds = new Set([
-      ...currentProductIds,
-      ...affectedProductIds,
-    ]);
-    for (const affectedProductId of affectedProductIds) {
-      if (currentProductIds.includes(affectedProductId)) {
+    const currentPriceIds = await this.subscribedPriceIds(uid);
+    let priorPriceIds = new Set([...currentPriceIds, ...affectedPriceIds]);
```

Comment:
You had asked me about code comments before -- maybe it's worth mentioning here or in the JSDoc-style comment at the top of this method that we do not know what the change is, because our DB updates and such occur in parallel, which is why we have to do the Set operation? You have a similar disclaimer for the `playUpdates` method below.

Also perhaps we could explain a little more here what `priorPriceIds` is intended to represent? I get what affected are -- those price ids from the subscription that changed in some way TBD. I get what current are -- those price ids the user is currently subscribed to. What is priorPriceIds? Those price ids the user is no longer subscribed to? Maybe we can change the local variable names here to be `activePriceIds` (instead of `currentPriceIds`) and `inactivePriceIds` (instead of `priorPriceIds`)?